src/libs/actions/Report.js

```diff
@@ -1635,6 +1635,79 @@ function leaveRoom(reportID) {
     navigateToConciergeChat();
 }
 
+/**
+ * Flag a comment as offensive
+ *
+ * @param {String} reportID
+ * @param {Object} reportAction
+ * @param {String} severity
+ */
+function flagComment(reportID, reportAction, severity) {
+    const newDecision = {
+        "decision": "pending",
+    };
+
+    const message = reportAction.message[0];
+    const reportActionID = reportAction.reportActionID;
+
+    const updatedDecisions = [...(message.moderationDecisions || []), newDecision];
+
+    const updatedMessage = {
+        ...message,
+        moderationDecisions: updatedDecisions,
+    };
+
+    const optimisticReportActions = {
+        [reportActionID]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+            message: [updatedMessage],
+        },
+    };
+
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`, value: optimisticReportActions,
+        },
+    ];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: {
+                [reportActionID]: {
+                    ...reportAction,
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+
+    const successData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: {
+                [reportActionID]: {
+                    pendingAction: null,
+                },
+            },
+        },
+    ];
+
+    const timestamp = Date.now();
```

Comment:
We don't need to pass the timestamp in from the front-end, we can generate it when we insert in the backend. Might have missed this in the auth review